File: app/src/main/java/net/radityalabs/contactapp/presentation/di/component/ContactDetailComponent.java
package net.radityalabs.contactapp.presentation.di.component;

import net.radityalabs.contactapp.presentation.di.module.ContactDetailModule;
import net.radityalabs.contactapp.presentation.ui.fragment.ContactDetailFragment;

import javax.inject.Singleton;

import dagger.Subcomponent;

/**
 * Created by radityagumay on 7/14/17.
 */

@Singleton
@Subcomponent(modules = ContactDetailModule.class)
public interface ContactDetailComponent {
    void inject(ContactDetailFragment fragment);
}

File: app/src/main/java/net/radityalabs/contactapp/presentation/di/component/ContactDetailComponent.kt
package net.radityalabs.contactapp.presentation.di.component

import net.radityalabs.contactapp.presentation.di.module.ContactDetailModule
import net.radityalabs.contactapp.presentation.ui.fragment.ContactDetailFragment

import javax.inject.Singleton

import dagger.Subcomponent

/**
 * Created by radityagumay on 7/14/17.
 */

@Singleton
@Subcomponent(modules = arrayOf(ContactDetailModule::class))
interface ContactDetailComponent {
    fun inject(fragment: ContactDetailFragment)
}
